### PR TITLE
Fixed duplication of messages

### PR DESCRIPTION
**Pull request by bilak**
The duplication of messages should be fixed by this implementation. However I'm not sure about [UUID performance](https://community.oracle.com/tech/developers/discussion/4496765/java-uuid-generation-performance-impact). If we need to improve that, let me know.

### CODE REVIEW

**Comment by efdknittlfrank** [comment]
Can you set the target branch to [eurofunk](https://github.com/Eurofunk/logback-elasticsearch-appender/tree/eurofunk)?

UUID shouldn't be a problem performance-wise on the client. I hope Elasticsearch isn't negatively affected by multiple documents with the same id and with a different id format (default id format seems to be "nBgY0IIBrz7ef8VblF5V" for ES 7.7 and "HYgZ0IIBBnMpP-r9y1D5" for ES 7.17).

Apart from that, it looks good and I'm fine with the change. Make sure to increase the artifact version before publishing (perhaps you can immediately increase it as part of this PR?)

**Review by efdknittlfrank — APPROVED** [review]

**Comment by efdknittlfrank** [comment]
Btw, the duplication is not really fixed, is it? It's just that duplicates overwrite the original log message (incrementing its _version field)

File: pom.xml [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
     <groupId>com.internetitem</groupId>
     <artifactId>logback-elasticsearch-appender</artifactId>
-    <version>1.6-eocs</version>
+    <version>1.7-eocs</version>
```
Comment (efdknittlfrank) [inline pom.xml]: Please make that `1.6-eocs-2` or `1.6-eocs.2` (we are still building on the 1.6 version of the upstream project).

(Yes, we should have started with `eocs-0`/`eocs.0` or `eocs-1`/`eocs.1` …)

**Review by efdknittlfrank — APPROVED** [review]

**Comment by bilak** [comment]
> Btw, the duplication is not really fixed, is it? It's just that duplicates overwrite the original log message (incrementing its _version field)

* if message was written to ES but there is some issue in request/response (eg res.code != 200) then it will be replaced/updated (_version is changed)
* if message wasn't written then it will be the new message identified by _id: UUID

Maybe we can even try [this one](https://stackoverflow.com/a/73475055/266254) but I'd say current implementation should be ok

**Pull request merged**